[zh]Sync reference-12 (#33827)
kubernetes/website: content/zh/docs/reference/kubectl/cheatsheet.md

echo "[[ $commands[kubectl] ]] && source <(kubectl completion zsh)" >> ~/.zshrc # 在您的 zsh shell 中永久的添加自动补全
```

<!--
### A Note on --all-namespaces
-->
### 关于 --all-namespaces 的一点说明

<!--
Appending `--all-namespaces` happens frequently enough where you should be aware of the shorthand for `--all-namespaces`:
-->
附加 `--all-namespaces` 经常发生，你应该知道 `--all-namespaces` 的简写：

```kubectl -A```

> 

Reviewer: How about this wording:

```
我們經常用到 `--all-namespaces` 參數，你應該要知道以下簡寫：
```

  && kubectl config use-context gce

kubectl config unset users.foo                       # 删除用户 foo

> # 设置/显示上下文/命名空间的短别名（仅适用于 bash 和 bash 兼容的 shell
> # 在使用 kn 设置命名空间之前要设置当前上下文） 

Reviewer: ```suggestion
# 設置或顯示 context / namespace 的短别名
# （仅适用于 bash 和兼容 bash 的 shell，在使用 kn 设置命名空间之前要先设置 current-context）
```

kubectl top pod POD_NAME --sort-by=cpu              # 显示给定 Pod 的指标并且按照 'cpu' 或者 'memory' 排序
```

<!--
## Copy files and directories to and from containers
-->
## 从容器中复制文件和目录

<!--
```bash
kubectl cp /tmp/foo_dir my-pod:/tmp/bar_dir            # Copy /tmp/foo_dir local directory to /tmp/bar_dir in a remote pod in the current namespace
kubectl cp /tmp/foo my-pod:/tmp/bar -c my-container    # Copy /tmp/foo local file to /tmp/bar in a remote pod in a specific container
kubectl cp /tmp/foo my-namespace/my-pod:/tmp/bar       # Copy /tmp/foo local file to /tmp/bar in a remote pod in namespace my-namespace
kubectl cp my-namespace/my-pod:/tmp/foo /tmp/bar       # Copy /tmp/foo from a remote pod to /tmp/bar locally
```
-->
```bash

> kubectl cp /tmp/foo_dir my-pod:/tmp/bar_dir            # 将 /tmp/foo_dir 本地目录复制到当前命名空间中远程 pod 中的 /tmp/bar_dir

Reviewer 2: ```suggestion
kubectl cp /tmp/foo_dir my-pod:/tmp/bar_dir            # 将 /tmp/foo_dir 本地目录复制到远程当前命名空间中 Pod 中的 /tmp/bar_dir
```

> kubectl cp /tmp/foo my-pod:/tmp/bar -c my-container    # 将 /tmp/foo 本地文件复制到特定容器中远程 pod 中的 /tmp/bar

Reviewer 2: ```suggestion
kubectl cp /tmp/foo my-pod:/tmp/bar -c my-container    # 将 /tmp/foo 本地文件复制到远程 Pod 中特定容器的 /tmp/bar 下
```

> kubectl cp /tmp/foo my-namespace/my-pod:/tmp/bar       # 将 /tmp/foo 本地文件复制到远程 “my-namespace” 命名空间中 pod 中的 /tmp/bar

Reviewer 2: ```suggestion
kubectl cp /tmp/foo my-namespace/my-pod:/tmp/bar       # 将 /tmp/foo 本地文件复制到远程 “my-namespace” 命名空间内指定 Pod 中的 /tmp/bar
```

> kubectl cp my-namespace/my-pod:/tmp/foo /tmp/bar       # 将 /tmp/foo 从远程 pod 复制到本地 /tmp/bar

Reviewer 2: ```suggestion
kubectl cp my-namespace/my-pod:/tmp/foo /tmp/bar       # 将 /tmp/foo 从远程 Pod 复制到本地 /tmp/bar
```

```

<!--
`kubectl cp` requires that the 'tar' binary is present in your container image. If 'tar' is not present,`kubectl cp` will fail.
For advanced use cases, such as symlinks, wildcard expansion or file mode preservation consider using `kubectl exec`.
-->
{{< note >}}

> `kubectl cp` 要求容器镜像中存在 'tar' 二进制文件。 如果 'tar' 不存在，`kubectl cp` 将失败。

Reviewer 2: ```suggestion
`kubectl cp` 要求容器镜像中存在 'tar' 二进制文件。如果 'tar' 不存在，`kubectl cp` 将失败。
```

> <!--
> `kubectl cp` requires that the 'tar' binary is present in your container image. If 'tar' is not present,`kubectl cp` will fail.
> For advanced use cases, such as symlinks, wildcard expansion or file mode preservation consider using `kubectl exec`.
> -->
> {{< note >}}
> `kubectl cp` 要求容器镜像中存在 'tar' 二进制文件。如果 'tar' 不存在，`kubectl cp` 将失败。
> 对于高级用例，例如符号链接、通配符扩展或文件模式保存，请考虑使用 `kubectl exec`。
> {{< /note >}}

Reviewer: ```suggestion
<!--
`kubectl cp` requires that the 'tar' binary is present in your container image. If 'tar' is not present,`kubectl cp` will fail.
For advanced use cases, such as symlinks, wildcard expansion or file mode preservation consider using `kubectl exec`.
-->
{{< note >}}
`kubectl cp` 要求容器镜像中存在 “tar” 二进制文件。如果 “tar” 不存在，`kubectl cp` 将失败。
对于進階用例，例如符号链接、通配符扩展或保留文件權限，请考虑使用 `kubectl exec`。
{{< /note >}}
```

<!--

* Read the [kubectl overview](/docs/reference/kubectl/) and learn about [JsonPath](/docs/reference/kubectl/jsonpath).

* See [kubectl](/docs/reference/kubectl/kubectl/) options.

* Also read [kubectl Usage Conventions](/docs/reference/kubectl/conventions/) to understand how to use kubectl in reusable scripts.

* See more community [kubectl cheatsheets](https://github.com/dennyzhang/cheatsheet-kubernetes-A4).
-->

> * 参阅 [kubectl 概述](/zh/docs/reference/kubectl/)，进一步了解[JsonPath](/zh/docs/reference/kubectl/jsonpath)。

Reviewer 2: ```suggestion
* 参阅 [kubectl 概述](/zh/docs/reference/kubectl/)，进一步了解 [JsonPath](/zh/docs/reference/kubectl/jsonpath)。
```
